src/LisExpr.chpl

```diff
@@ -12,8 +12,13 @@ module LisExpr
     
     type Symbol = string;
 
+    class ListClass {
+      type etype;
+      forwarding var l: list(etype);
+    }
+    
     /* type: list of genric list values */
-    type GenList = list(owned GenListValue);
+    type GenList = shared ListClass(owned GenListValue);
```

Comment:
We could use `owned` here, and that is probably the better choice. The one case here where it would matter (I think) is in the initializer, where we would need to deep copy, whereas the current code is storing a pointer to the existing object, which probably is not what we want now that I think about it. I have updated the code.